### PR TITLE
Feature: Use GM2Calc to calculate a_mu in the 2HDM

### PR DESCRIPTION
**Pull request by Expander**
Todo:
- [x] write GM2Calc interface for FS
- [x] actually use GM2Calc in 2HDM-like models

### CODE REVIEW

**Comment by Expander** [comment]
I believe this PR is ready to be merged.

Btw., we now have a nice cross-check between the FS-generated 1-loop expression and the one implemented in GM2Calc.

File: model_files/THDMII/LesHouches.in.THDMII [hunk 1]
```diff
@@ -16,7 +16,7 @@ Block FlexibleSUSY
    12   0                    # force output
    13   1                    # Top pole mass QCD corrections (0 = 1L, 1 = 2L, 2 = 3L)
    14   1.000000000e-11      # beta-function zero threshold
-   15   0                    # calculate observables (a_muon, ...)
+   15   1                    # calculate observables (a_muon, ...)
```
Comment (wkotlarski) [inline model_files/THDMII/LesHouches.in.THDMII:19]: I remember that @pathron had some objection to enabling calculation of observables by default. I can't remember why though.

**Comment by wkotlarski** [comment]
@Expander, I'm slightly confused. I though that `GM2Calc` works only in `NoFV` models (where fermion flavours are not in a multiplet). But the `THDM` we have is not like that.

**Comment by Expander** [comment]
Thanks for looking into this!

> @Expander, I'm slightly confused. I though that `GM2Calc` works only in `NoFV` models (where fermion flavours are not in a multiplet). But the `THDM` we have is not like that.

Yes, the MSSM calculation of GM2Calc works only for NoFV models, which means that the models must have conservation of the sfermion flavour (not fermion flavour). In the THDM this restriction is irrelevant, as there are no sfermions (and fermion flavour violation is allowed). :)

File: model_files/THDMII/LesHouches.in.THDMII (hunk 1 above)
Comment (Expander) [reply model_files/THDMII/LesHouches.in.THDMII]: You are right, we should probably not calculate all observables by default. I've just reverted this change.

**Review by wkotlarski — APPROVED** [review]

**Comment by pathron** [comment]
Yes the NoFV notation is very confusing!  What it really means is no familiy mixing in fermion sector (ie follows the slha 1 convention for sfermions).

**Comment by Expander** [comment]
> Yes the NoFV notation is very confusing! What it really means is no familiy mixing in fermion sector (ie follows the slha 1 convention for sfermions).

Sorry, I think there is a typo here: You probably mean "[it] means is no familiy mixing in **s**fermion sector". :)

**Comment by pathron** [comment]
:shakes head: sorry yes that is what I meant! I think i was cursed at birth to constantly make typos that completely reverse what i am saying...

**Comment by Expander** [comment]
> > Yes the NoFV notation is very confusing! What it really means is no familiy mixing in the sfermion sector (ie follows the slha 1 convention for sfermions).
> 
> But not always, right? For example in the `MSSMNoFV` SM fermions also don't mix. That's why we had to disable `FlexibleDecay` in this model.

You are right, in the MSSMNoFV the fermions don't mix. However, this is not a restriction regarding GM2Calc (and can in principle be lifted). The restriction in the MSSM is that sfermion families don't mix.

**Comment by pathron** [comment]
I think its worth discussing this.  

First the NoFV tag was really introduced because in some models we wanted family eigenstates, because that's how slha1 works.  They have 2 mass ordered stops, two mass ordered sbottoms, two mass ordered sleptons, and then just left and right states that don't mix at all for the first and second generation sfermions.  So many codes that expect slha1 output want something like this.  Maybe we should at least use NoSFV or NoSfermFV instead of NoSV.  It's still a bit confusing though because people often assume that the other models without this label treat FV properly using CKM and superCKM, but actually many of them don't they just permit one to have explicit FV soft masses.  

Actually even worse we have the CMSSM where the universal boundary conditions don't allow explicit sfermion violation. There we have no FV in the sfermion sector at all, because we don't add CKM and SuperCKM.  That is instead done in CMSSMCKM.  I don't like this.   I really think we could consider having a policy of using CKM and SuperCKM in all models that do not have NoFV, but I think that's what I proposed before and Alex objected?  The argument against that I recall is it complicates the model file making them less useful as simple examples to the user for how to write BCs etc.  That's true but it makes them more useful as models a user can just run!  Also we could leave the NoFV ones with the simpler BCs for more simple pedagogical examples.

Regarding the fermion FV, I guess you mean that in some model files we have split the fermions by generation and not allowed them to mix at the SARAH level.   I guess then we could also call these NoFermFV.  Again there's the potential issue that unless we use CKM we are often numerically making the Yukawas diagonal anyway....

**Pull request merged**